pytorch_lightning/loops/base.py

```diff
@@ -56,8 +56,8 @@ def trainer(self) -> Optional['pl.Trainer']:
 
     @trainer.setter
     def trainer(self, trainer: 'pl.Trainer'):
-        """Connect the Trainer to this loop and all children."""
-        if not isinstance(trainer, pl.Trainer) and trainer is not None:
+        """Connect the Trainer to itself and all its children loops"""
```

Comment:
"Connect the Trainer to itself" doesn't sound right

```suggestion
        """Connects this loop's trainer and it's children"""
```